src/main/java/com/powsybl/networkconverterserver/NetworkConverterService.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.networkconverterserver;
+
+import com.powsybl.commons.datasource.ReadOnlyMemDataSource;
+import com.powsybl.networkconverterserver.dto.NetworkIds;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.network.store.client.NetworkStoreClient;
+import com.powsybl.network.store.client.NetworkStoreService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.*;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.DefaultUriBuilderFactory;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * @author Abdelsalem Hedhili <abdelsalem.hedhili at rte-france.com>
+ */
+
+@Service
+@ComponentScan(basePackageClasses = {NetworkStoreClient.class})
+public class NetworkConverterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkConverterService.class);
+    private String caseServerBaseUri;
+    private String geoDataBaseUri;
+    private RestTemplate caseServerRest;
+    private RestTemplate geoDataRest;
+    private static final  String IIDM_GEO_DATA_API_VERSION = "v1";
+
+    @Autowired
+    private NetworkStoreService networkStoreService;
+
+    @Autowired
+    public NetworkConverterService(@Value("${backing-services.case.base-uri:http://case-server/}") String caseServerBaseUri,
+                                   @Value("${backing-services.geo-data.base-uri:http://geo-data-server/}") String geoDataBaseUri) {
+        RestTemplateBuilder restTemplateBuilder = new RestTemplateBuilder();
+        caseServerRest = restTemplateBuilder.build();
+        caseServerRest.setUriTemplateHandler(new DefaultUriBuilderFactory(caseServerBaseUri));
+        this.caseServerBaseUri = caseServerBaseUri;
+
+        restTemplateBuilder = new RestTemplateBuilder();
+        geoDataRest = restTemplateBuilder.build();
+        geoDataRest.setUriTemplateHandler(new DefaultUriBuilderFactory(geoDataBaseUri));
+        this.geoDataBaseUri = geoDataBaseUri;
+    }
+
+    NetworkIds persistentStore(String caseName) {
+        byte[] networkByte = getCaseAsByte(caseName);
```

Comment:
That would be really better if you can stream the data from the case store to the network importer without intermediate buffer. Let's keep it as it is for this PR but you can create a new issue to remind us to work on it just after.